src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -445,6 +450,11 @@ func (s *service) Start(ctx context.Context, r *taskAPI.StartRequest) (_ *taskAP
 
 	start := time.Now()
 	defer func() {
+		// if sandbox/container is not found in path
+		// it means that it is already deleted
+		if err != nil && isNotFound(err) {
```

Comment:
This may make sense in the kill/cleanup case, but in case of start/exec, if the container is not found or there is an error case, the error needs to be propogated. Setting the err to nil here will hide errors during container lifecycle.

Reply:
makes sense, updated